packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -64,7 +80,7 @@ export class GOVUKFrontendComponent {
 
 /**
  * @typedef ChildClass
- * @property {string} [moduleName] - The module name that'll be looked for in the DOM when initialising the component
+ * @property {string} moduleName - The module name that'll be looked for in the DOM when initialising the component
```

Comment:
**question** Do we need to do this because the `typeof moduleName` comes later in the constructor than the throwing of the `ElementError` which requires `component` to have a `moduleName` attribute? 

If that's the case, it could be a sign we need to reorder some code in the constructor (eg. throw an `InitError` early) 🤔

Reply:
If the `InitError` is moved before the `ElementError`, there's still complaining about the `ElementError` not having a valid component because `ElementError` expects a component with a defined `moduleName`. This is the case even if throwing the `ElementError` is wrapped in a if statement that checks that type of `moduleName` is not `undefined`.